jwst/photom/photom.py

```diff
@@ -1097,52 +1098,62 @@ def save_area_info(self, ftab, area_fname):
         if area_fname is not None and area_fname != "N/A":
             pix_area = datamodels.open(area_fname)
 
-            # Copy the pixel area data array to the appropriate attribute
-            # of the science data model
             if self.instrument != 'NIRSPEC':
+                area_ster, area_a2 = None, None
+                try:
+                    # Load the average pixel area values from the AREA reference file header
+                    # Don't need to do this for NIRSpec, because pixel areas have already
+                    # been copied using save_area_nirspec
+                    if isinstance(self.input, datamodels.MultiSlitModel):
+                        # Note that this only copied to the first slit.
+                        self.input.slits[0].area = pix_area.data
+                    else:
+                        ystart = self.input.meta.subarray.ystart - 1
+                        xstart = self.input.meta.subarray.xstart - 1
+                        yend = ystart + self.input.meta.subarray.ysize
+                        xend = xstart + self.input.meta.subarray.xsize
+                        self.input.area = pix_area.data[ystart: yend,
+                                                        xstart: xend]
+                    log.info('Pixel area map copied to output.')
+
+                    area_ster = pix_area.meta.photometry.pixelarea_steradians
+                    if area_ster is None:
+                        log.warning('The PIXAR_SR keyword is missing from %s', area_fname)
+                    area_a2 = pix_area.meta.photometry.pixelarea_arcsecsq
+                    if area_a2 is None:
+                        log.warning('The PIXAR_A2 keyword is missing from %s', area_fname)
+                    log.info('PIXAR_SR and PIXAR_A2 values obtained from AREA reference file.')
+
+                except AttributeError:
+                    # The area reference file might be older, try the photom reference file
+                    area_ster = ftab.meta.photometry.pixelarea_steradians
+                    if area_ster is None:
+                        log.warning('The PIXAR_SR keyword is missing from %s', ftab.meta.filename)
+                    area_a2 = ftab.meta.photometry.pixelarea_arcsecsq
+                    if area_a2 is None:
+                        log.warning('The PIXAR_A2 keyword is missing from %s', ftab.meta.filename)
+                    log.info('Values for PIXAR_SR and PIXAR_A2 obtained from PHOTOM reference file.')
```

Comment:
Same comment here regarding this message only getting emitted if both are not None.